Initial Draft

btat opened the pull request.

btat committed:
Initial Draft

btat commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+[id='sec.2_0.issue']
+==== Known Issues
+
```
Add note about values-file change (1.x vs 2.x format)

btat replied:
Mention nfs broker is not available for 2.0

btat replied:
Mention removal of cf-usb

btat commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,115 @@
+
+[id='sec.2_0.issue']
+==== Known Issues
+
```
Add note about migration

gaktive replied:
Advise 1.5.2 admins to read the link to the pertinent section in the CAP docs.

btat reviewed (commented):
Maybe add a section for deprecations

gaktive commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,115 @@
+** bpm: 1.1.6
+** capi: 1.89.0
+** cf-acceptance-tests: 12.17
+** cf-deployment: 12.17
```
Per https://github.com/cloudfoundry-incubator/kubecf/blob/master/dependencies.yaml, this is 12.36 for release. I'll see if I can dig up the other bits of info via the role manifest

gaktive replied:
https://github.com/cloudfoundry-incubator/kubecf/blob/master/deploy/helm/kubecf/values.yaml has some versions that override others from the deployment itself

gaktive replied:
Upstream is our base reference: https://github.com/cloudfoundry/cf-deployment/blob/v12.36.0/cf-deployment.yml#L2344-L2468 so that's the 1st step. Helm chart values then overrides some of those.

gaktive committed:
first pass at component version updates

Relying on helm chart values along with cf upstream version. Question marks around variables that need clarification based on presence, absence or wildly different version number that previous scf value.

gaktive reviewed (commented):
In progress. 
* Need to sort out values for `bpm`, `groot-btrfs`, `mapfs` & `nfs-volume` as they aren't in values.
* `pxc` looks like a downgrade, so I need to review semver
* Do we need to mention components like `brains` & `bosh-dns-aliases`?
* Confirm removal of `mysql` & `sle12`

Commit:
removed nfs-volume, mapfs & pxc

Our pxc version is based on docker, not upstream, so we moved away from the upstream versions. If needed, our pxc is 0.9.4, which maps to Percona-XtraDB-Cluster-5.7.28-31.41.

gaktive commented:
`bpm` checks out

Commit:
tidy on components and bumped kubecf to 2.3.0

in anticipation of RC3. need to confirm if operator bump needed or not

btat committed:
Add deprecated components and update sample-values file

btat committed:
Remove extra word

btat committed:
Add additional items

gaktive reviewed (approved):
Good start. Let's see what else bubbles up from testing.

Commit:
tidied up buildpack versions

4th ordinal is internal to SUSE.

Commit:
SLE12 and cflinuxfs2 are no longer part of CAP

Ties into a deprecation announcement from the CAP 1.5 release notes. We had warned people that this was coming; now it has happened.

Commit:
clarified stack changes and added pxc wordily

btat committed:
Add Azure Broker limitations and Eirini stacks info

btat committed:
Add autoscaler workarounds

btat committed:
Fix version number

btat committed:
Add entry for new in Stratos

Commit:
added cf-usb removal to known issues

btat committed:
Add link to migration in docs

btat committed:
Merge branch 'release-2.0' of ssh://github.com/suse/cap-release-notes into release-2.0

btat committed:
Fix entity syntax and name

satadruroy commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+
+[id='sec.2_0.new']
+==== What Is New?
+* {kubecf} 2.2.2 replaces {scf}:
```
Should we expand on this, as it is it doesn't really tell the user anything about how different KubeCF is from SCF. We can say something like, CAP 2 is based on KubeCF 2.2.2 which uses quarks operator to deploy releases from cf-deployment into Kubernetes environments. Maybe check with @troytop for the wording above.

gaktive replied:
That copy looks like a good start. Maybe we can link to that Github project and its release notes for additional context, similar to what we do for Stratos. We should also talk more about the need for the operator and link to that project.

troytop replied:
Agreed that we need more descriptive content at the start of the doc. We shouldn't block this PR on its absence, but we should add more as soon as possible. I can help with this.

satadruroy reviewed (commented):
Nothing major. Haven't checked the versions but LGTM otherwise.

troytop commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+
+* Eirini will only work on a cluster that has `.cluster.local` as the cluster domain
+* When Eirini is enabled, both `features.suse_default_stack` and `features.suse_buildpacks` must be enabled as well. A cflinuxfs3 Eirini image is currently not available, and the SUSE stack must be used. By default, both features are enabled and default.
+* SLE12 and cflinuxfs2 are no longer supported stacks. They have been deprecated from CAP 1.x.
```
suggest: "SLE12 and cflinuxfs2 have been removed and are no longer supported."
This is one of the more important things in the release notes. We should also link to any docs we have (or previous release notes in 1.x about deprecation) that provide advice on migrating to the newer stacks.

troytop reviewed (changes requested):
Would like to see a bit more emphasis on the removal of the old stacks.

prabalsharma commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+[IMPORTANT]
+====
+* Autoscaler can go into a `CrashLoopBackoff` state when enabled using `helm upgrade` on systems where the DNS entries have already been set up. The `asactors` pod can be deleted to recover from this state as kubernetes will re-create the pod
+
```
Do we want to mention this issue as well: https://github.com/cloudfoundry-incubator/kubecf/issues/973 ?
Looks like we don't have a workaround for this one.

satadruroy replied:
Yes, @mook-as has confirmed this issue can trigger due to what appears to be a race condition that only happens during an upgrade. The workaround would be to not use the `helm upgrade` but to turn on autoscaler during the initial deployment. So, we should update above to read:

`Occasionally, autoscaler database pod can go into a `CrashLoopBackoff` when Autoscaler is enabled via a `helm upgrade`. It is recommended Autoscaler is deployed along with other CAP 2 components during the initial `helm install`.  `

`

prabalsharma commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+
+====
+
+* Eirini will only work on a cluster that has `.cluster.local` as the cluster domain
```
What does this mean? Eirini works with LB

btat replied:
This was listed in the Jira for release notes collection. I believe it's referring to `--cluster-domain=<default-local-domain>` described at https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/#introduction. This doesn't affect public clouds since that should already be the default. Will expand to clarify.

prabalsharma commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+* The Open Service Broker for Azure is only compatible with Kubernetes 1.15 or earlier.
+* The `cf-usb` service brokers from CAP 1.x will not work with CAP 2.0, along with Kubernetes 1.16 or higher, so they are no longer supported. As noted in the Features & Fixes above, please migrate over to either Minibroker or an OSBAPI-compliant broker available via your public cloud platform.
+* During the Open Service Broker for Azure set up process, the svc/catalog chart install will encounter a `OOMKilled` state and fail. The controllerManager's requests and limits for the CPU and memory must be increased to avoid this. As an example, increasing these values to double the default will allow for a successful installation.
++
```
Eirini deployment requires proper cleanup of CSR: https://github.com/cloudfoundry-incubator/kubecf/issues/877

btat replied:
@prabalsharma can you expand on what you mean by proper cleanup?

prabalsharma replied:
`csr` for bits is not namespaced, so just a `helm delete` of kubecf won't get rid of it.
If the cluster had a previous deployment of eirini, one will have to delete bits `csr` manually, before a new deployment.

gaktive commented:
We may need an entry talking about external UAA in 2.0.

prabalsharma commented on adoc/values-sample.yaml:
```diff
@@ -1,1178 +1,482 @@
+      # crt: ~
+      # key: ~
+    # annotations: {}
+    # labels: {}
```
This is outdated. Missing
```
suse_default_stack:
    enabled:  true
```

prabalsharma commented on adoc/release-2.0.adoc:
```diff
@@ -0,0 +1,136 @@
+====
+
+* Eirini will only work on a cluster that has `.cluster.local` as the cluster domain
+* When Eirini is enabled, both `features.suse_default_stack` and `features.suse_buildpacks` must be enabled as well. A cflinuxfs3 Eirini image is currently not available, and the SUSE stack must be used. By default, both features are enabled and default.
```
Needs rephrasing.


```
When Eirini is enabled, both `features.suse_default_stack` and `features.suse_buildpacks` must be enabled. ~as well~ . ...
... By default, both features.suse_default_stack and features.suse_buildpacks are enabled.`
```

btat replied:
Just to confirm, you mean to drop `as well`? Enclosing words with tildes make it a subscript.

btat committed:
Updates according to feedback

btat committed:
Remove incorrect statement, will only be default in 2.3+

btat committed:
Clarify Eirini deployment CSR cleanup

btat committed:
Fix incorrect kube reqs link

Commit:
Update Travis config to use new container

ghost commented:
Is it intended for this to continue life as an unresolved PR? Or can it just be closed at this point? (It should be separate from the 1.x content anyway.)

_If you do merge this to master,_ be aware that the CAP 1.x release notes are currently rebuilt from master via IBS and you will need to change the branch referenced in the IBS package's _service file to some 1.x maintenance branch. Otherwise, the 2.0 content gets published as 1.x content. Also note that the 2.0 package is built off this branch, release-2.0.
